### ISSUE TITLE
When creating new segment, intelligently select tag from segment tag list.

### ISSUE THREAD

**Issue by isolver** [isolver/OpenHandWrite#18]
MarkWrite has a list of 'predefined' segment names. See issue #10 for details on how this list is created.

The list is used in the 'Create Segment Dialog' text entry dropdown. However there is no default segment name selection logic in place that selects a suggested segment name from the list based on previously selected list index or such.

So current state of app in regard to the usage of this list is:
- no tracking is done of what the last selected index in the segment name list was
- segment name in create segment dialog does not default to one of the predefined names in the list
#### TODO:

Should add option for app to set a new segments name to an initial value based on the last created segments name (and / or predefined name list index). 

Not sure exactly how this logic should work though, seems unclear to me right now:
- When should a new segment be given an initial name from the predefined segment list and when should no initial name be set.
- In cases where an initial segment name should be set, what are the details for the logic needed to decide what index should be used from the predefined segment name list.

Factors like user not using auto selected segment name and using some custom name instead need to be considered.

Maybe I am overcomplicating what needs to be done here?

**Comment by Mark-Torrance** [comment]
I suggest:
- System level setting (with easy access - so set from a button rather than buried in options) for suggest / don't suggest.
- if don't suggest then current behaviour (quick search: starting to type pulls up fist item in list that matches typed characters, I think)
- if suggest then disable quick search and display the item in the tag list that has index n+1 where n is the index of the last item chosen. But with the option to reject this by either typing in a different tag, and or choosing a different tag from the dropdown list.

I think this is the simplest solution for now.

**Comment by isolver** [comment]
On Fri, Apr 10, 2015 at 4:47 AM, Mark-Torrance notifications@github.com
wrote:

> if suggest then disable quick search and display the item in the tag list
> that has index n+1 where n is the index of the last item chosen. But with
> the option to reject this by either typing in a different tag, and or
> choosing a different tag from the dropdown list.

In this second mode, can the user type / select a tag that is not in the
predefined tag list and must use a tag from the list? If they can still
enter 'undefined tags' that are not in the list, what should happen in
terms of:

>  display the item in the tag list that has index n+1 where n is the index
> of the last item chosen.

since no last item was chosen?

**Comment by Mark-Torrance** [comment]
In the Suggest mode they can do any of:
- accept suggestion
- choose a different word from the list
- type in new tag

The index of the next suggestion...
- accept suggestion: n+1 where n is index of suggestion
- choose a different word from the list: n+1 where n is the index of the chosen word 
- type in new tag: n where n is the index of the suggestion

I think that works.

**Comment by isolver** [comment]
> The index of the next suggestion...
> ...
> - type in new tag: n where n is the index of the suggestion

To clarify this case, is the following true:
1. the suggestion index has not changed from the last segment creation
2. so we select the same index as was previously suggested (but was rejected by user in favor of custom tag)

**Comment by Mark-Torrance** [comment]
Yes. So if they just keep on typing their own stuff in they will always be offered the same suggestion. The idea is that if there is a mismatch between tag list and what they see they can type in one or more tags and then find their place in the list again.

**Comment by isolver** [comment]
Another things to confirm please: 
1. The predefined segment name list can have duplicate text entries in it. i.e. the word 'Word' can be in the predefined segment tag list > 1 time.
2. When parsing a data files associated segment tag file, any spaces and newline characters are striped from the beginning and end of each tag.

**Comment by Mark-Torrance** [comment]
The predefined segment name list can have duplicate text entries in it. i.e. the word 'Word' can be in the predefined segment tag list > 1 time.
    YES. Obviously original order of words (or whatever) needs to be retained in the parsed taglist.

When parsing a data files associated segment tag file, any spaces and newline characters are striped from the beginning and end of each tag.
    YES - rstrip and lstrip all unprintables (so also tabs).

**Comment by Mark-Torrance** [comment]
This is needed to properly exploit the tag list.  However no time to do this for first release. Therefore demote.
